From 53d72bca75a2bfc75bceea0d094131c9d53bd942 Mon Sep 17 00:00:00 2001
From: "Frank Ch. Eigler" <fche@redhat.com>
Date: Fri, 28 Mar 2014 15:24:19 -0400
Subject: [PATCH] PR16766 cont'd: unbreak --runtime=dyninst

* translate.cxx: While emitting systemtap_module_refresh(), protect
  the printk diagnostics with #if defined(__KERNEL__).
---
 translate.cxx | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/translate.cxx b/translate.cxx
index 17dedd4..90b04d8 100644
--- a/translate.cxx
+++ b/translate.cxx
@@ -1887,7 +1887,9 @@ c_unparser::emit_module_refresh ()
   o->newline() << "int state = atomic_read (session_state());";
   o->newline() << "if (state != STAP_SESSION_RUNNING && state != STAP_SESSION_STARTING) {";
   // cannot _stp_warn etc. since we're not in probe context
-  o->newline(1) << "printk (KERN_ERR \"stap module notifier triggered in unexpected state %d\", state);";
+  o->newline(1) << "#if defined(__KERNEL__)";
+  o->newline() << "printk (KERN_ERR \"stap module notifier triggered in unexpected state %d\", state);";
+  o->newline() << "#endif";
   o->newline() << "return;";
   o->newline(-1) << "}";
 
-- 
1.8.3.1

